### ISSUE TITLE
Share Your Coding Problems Here!

### ISSUE THREAD

**Issue by ShiyangLai** [KnowledgeLab/Thinking-With-Deep-Learning-2025#3]
Hi students!

This issue is a place for you to share any coding problems you're encountering.

Please provide:
- A brief description of your problem.
- Any relevant code snippets.
- Error messages or logs (if applicable).
- What you've tried so far.

Feel free to respond to each other's comments to help out! Let's collaborate and solve problems together ✊

**Comment by ulisolovieva** [comment]
HW1.2 

- Titanic dataset link has expired, alt: https://hbiostat.org/data/repo/titanic3.csv
- Boston dataset is no longer available with the current scikit-learn version, but can be loaded in manually

Question: Why are the keras and pytorch deep NNs (=<.7;  not the experimental poisson & adagrad examples) performing worse on accuracy than the shallow NNs (>.9)? I’ve rerun my code a couple of times and the pattern stays.

**Comment by ana-yurt** [comment]
HW1.2 Introducing Deep Learning
- In the section on LayerConductance for interpretability, I am a bit unclear why we apply the function on the sigmoid layer instead of the hidden layer. What is the intuition here?

**Comment by psymichaelzhu** [comment]
In HW2 module1 Experiment 2 illustration, we attempted to build a wide-deep network.
I noticed that in the implementation, it seems that the wide part did not use sparse one-hot encoding. Instead, it used the embedding representation (see below). 
I'm wondering if we should use one-hot encoding?
```
 # Wide part: Sparse representation
self.wide_embeddings = create_embeddings(categorical_cardinalities)
wide_input_dim = num_numeric_features + sum(emb.embedding_dim for emb in self.wide_embeddings)
```
